pkg/ccl/changefeedccl/encoder_avro.go

```diff
@@ -220,14 +220,21 @@ func (e *confluentAvroEncoder) EncodeValue(
 	v, ok := e.valueCache.Get(cacheKey)
 	if ok {
 		registered = v.(confluentRegisteredEnvelopeSchema)
-		if err := registered.schema.after.refreshTypeMetadata(updatedRow); err != nil {
-			return nil, err
-		}
 		if prevRow.IsInitialized() && registered.schema.before != nil {
 			if err := registered.schema.before.refreshTypeMetadata(prevRow); err != nil {
 				return nil, err
 			}
 		}
+		if registered.schema.after != nil {
+			if err := registered.schema.after.refreshTypeMetadata(updatedRow); err != nil {
+				return nil, err
+			}
+		}
+		if registered.schema.record != nil {
+			if err := registered.schema.record.refreshTypeMetadata(updatedRow); err != nil {
+				return nil, err
+			}
+		}
```

Comment:
I don't see this addition of the refresh on `schema.record` mentioned in the commit message. Mind giving me some context on this bit?

Reply:
Originally the code assumed that `schema.after` would always be set so that's why there was no nil check on it. In c00bf0feb783021b76893b905fb19f8ec4e222bc, this was changed so that in some cases we would set `schema.record` instead of `schema.after`. I believe it would match the original intent of the code to call `refreshTypeMetadata` on whichever field is set.

Reply:
I updated the commit message to following (lmk if anything is missing or sounds wrong): 

Previously, the "after" field was populated across all envelope formats.
However, in commit c00bf0f, row data was moved to the “after” field for envelope
= wrapped and to “record” field for other envelope formats. But the commit did
not add a nil check for registered.schema.after which is only populated for
wrapped envelope. `registered.schema.after` is now nil for other envelope
formats, and calling `refreshTypeMetadata` on it could lead to node panics. In
addition, this commit did not add the logic calling `refreshTypeMetadata` on the
registered.schema.record for other envelope formats.  This patch addresses the
issue by adding a defensive nil check before invoking `refreshTypeMetadata` and
adding `refreshTypeMetadata` for the record field when it's set to match the
original intents of the commit.

Reply:
Thanks! I should have checked the original PR where this was already discussed.

Reply:
Nah, I should've updated the commit description. Thanks for the thorough PR review!